consensus/src/execution_pipeline.rs

```diff
@@ -233,7 +240,16 @@ impl ExecutionPipeline {
                 .expect("Failed to spawn_blocking().")
             }
             .await;
-            let pipe_line_res = res.map(|output| PipelineExecutionResult::new(input_txns, output));
+            let (commit_tx, commit_rx) = oneshot::channel();
+            let pipe_line_res =
+                res.map(|output| PipelineExecutionResult::new(input_txns, output, commit_rx));
+            pre_commit_tx
+                .send(PreCommitCommand {
+                    block_id,
+                    parent_block_id,
+                    result_tx: commit_tx,
+                })
+                .expect("Failed to send block to pre_commit stage.");
```

Comment:
Shall we skip `pre_commit` if the execution result `is_err()`?

Reply:
makes sense